tokio/src/task/join_set.rs

```diff
@@ -73,6 +92,33 @@ impl<T> JoinSet<T> {
 }
 
 impl<T: 'static> JoinSet<T> {
+    /// Returns a [`Builder`] that can be used to configure a task prior to
+    /// spawning it on this `JoinSet`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::task::JoinSet;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let mut set = JoinSet::new();
+    ///
+    ///     // Use the builder to configure a task's name before spawning it.
+    ///     set.build_task()
+    ///         .name("my_task")
+    ///         .spawn(async { /* ... */ });
+    /// }
+    /// ```
+    #[cfg(all(tokio_unstable, feature = "tracing"))]
+    #[cfg_attr(docsrs, doc(cfg(all(tokio_unstable, feature = "tracing"))))]
+    pub fn build_task(&mut self) -> Builder<'_, T> {
```

Comment:
Idiom wise, Tokio favors initializing the builder (vs a `build` method on the type being built).

So, this would be:

```
Builder::new(&mut join_set)`
```

Reply:
We don't have builders whose constructor takes arguments, so I don't think we really have an idiom for this. A `build` method seems more convenient to use.

Reply:
Seconded, for builder types that need to borrow some sort of context, it works much better with a fluent API flow like this.